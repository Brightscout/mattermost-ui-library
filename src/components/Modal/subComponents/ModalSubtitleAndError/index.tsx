import React from 'react';

type ModalSubtitleAndErrorProps = {
    subTitle?: string | JSX.Element;
    error?: string | JSX.Element;
}

const ModalSubtitleAndError = ({
    subTitle,
    error,
}: ModalSubtitleAndErrorProps) : JSX.Element => (
    <>
        {subTitle && <p className='modal__subtitle margin-0 margin-bottom-15'>{subTitle}</p>}
        {error && <p className='modal__error margin-left-15 margin-bottom-10 error-text text-capitalize'>{error}</p>}
    </>
);

export default ModalSubtitleAndError;
