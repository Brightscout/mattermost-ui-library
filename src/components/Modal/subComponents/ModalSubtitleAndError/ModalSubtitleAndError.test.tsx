import React from 'react';
import {shallow} from 'enzyme';

import ModalSubtitleAndError from '.';

// Mock props data
const subTitle = 'Modal subtitle';
const error = 'Modal error';

describe('ModalSubtitleAndError', () => {
    it('Should render correctly and matches snapshot when only the subTitle is passed', () => {
        const component = shallow(<ModalSubtitleAndError subTitle={subTitle}/>);
        expect(component).toMatchSnapshot();
        expect(component.text().includes(subTitle)).toBeTruthy();
    });

    it('Should render correctly and matches snapshot when only the error is passed', () => {
        const component = shallow(<ModalSubtitleAndError error={error}/>);
        expect(component).toMatchSnapshot();
        expect(component.text().includes(error)).toBeTruthy();
    });

    it('Should render correctly and matches snapshot when both subTitle and error props are passed', () => {
        const component = shallow(
            <ModalSubtitleAndError
                subTitle={subTitle}
                error={error}
            />,
        );
        expect(component).toMatchSnapshot();
        expect(component.text().includes(error)).toBeTruthy();
    });
});
