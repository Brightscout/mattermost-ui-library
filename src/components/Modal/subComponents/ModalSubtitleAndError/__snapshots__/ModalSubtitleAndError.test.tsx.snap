// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSubtitleAndError Should render correctly when both subTitle and error props are passed 1`] = `
<Fragment>
  <p
    className="modal__subtitle margin-0 margin-bottom-15"
  >
    Modal subtitle
  </p>
  <p
    className="modal__error margin-0 margin-bottom-10 error-text text-center text-capitalize"
  >
    Modal error
  </p>
</Fragment>
`;

exports[`ModalSubtitleAndError Should render correctly when only the error is passed 1`] = `
<Fragment>
  <p
    className="modal__error margin-0 margin-bottom-10 error-text text-center text-capitalize"
  >
    Modal error
  </p>
</Fragment>
`;

exports[`ModalSubtitleAndError Should render correctly when only the subTitle is passed 1`] = `
<Fragment>
  <p
    className="modal__subtitle margin-0 margin-bottom-15"
  >
    Modal subtitle
  </p>
</Fragment>
`;
